### PR TITLE
Fix scheduler tests with tracking and failure messages

### LINKED ISSUE #6789: Add run_done_callback to scheduler
The "run_done_callback" is in `job.py` but only as an MVP. The output from `forward_model_ok` is not handled correctly yet.

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #6789 and #6773 

**Approach**
See commit messages.

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/scheduler/job.py [hunk 1]
```diff
@@ -57,7 +61,9 @@ def __init__(self, scheduler: Scheduler, real: Realization) -> None:
         self.started = asyncio.Event()
         self.returncode: asyncio.Future[int] = asyncio.Future()
         self.aborted = asyncio.Event()
-        self._scheduler = scheduler
+        self._scheduler: Scheduler = scheduler
+        self._callback_status_msg: str = ""
+        self._max_submit: int = 2
```
Comment (xjules) [inline src/ert/scheduler/job.py:66]: Why is `_max_submit` set as an attribute here? I think this one was suppose to just come as function parameter for `__call__`?

File: src/ert/scheduler/job.py [hunk 2]
```diff
@@ -87,12 +93,22 @@ async def _submit_and_run_once(self, sem: asyncio.BoundedSemaphore) -> None:
...
+                if self._callback_status_msg != "":
+                    self._callback_status_msg = status_msg
+                else:
+                    self._callback_status_msg += (
+                        f"\nstatus from done callback: {status_msg}"
+                    )
+
+                if callback_status == LoadStatus.LOAD_SUCCESSFUL:
+                    await self._send(State.COMPLETED)
+                elif callback_status == LoadStatus.TIME_MAP_FAILURE:
```
Comment (xjules) [inline src/ert/scheduler/job.py:107]:
```python
else:
   await self._send(State.FAILED)
```

File: src/ert/scheduler/job.py [hunk 3]
```diff
@@ -111,6 +127,7 @@ async def _submit_and_run_once(self, sem: asyncio.BoundedSemaphore) -> None:
     async def __call__(
         self, start: asyncio.Event, sem: asyncio.BoundedSemaphore, max_submit: int = 2
     ) -> None:
+        self._max_submit = max_submit
```
Comment (xjules) [inline src/ert/scheduler/job.py:130]: Why an attribute?

Comment (berland) [reply src/ert/scheduler/job.py:130]: It is only for the sake of the error message. The class `Job` does not hold the config, so it needs to keep track of `max_submit` from `__call__` in order to know how to make the error message. The attribute name can be anonymized further at least.

File: src/ert/scheduler/job.py (hunk 2 above)
Comment (berland) [reply src/ert/scheduler/job.py:107]: will fix!

Looking forward to 3.10:
```
                match callback_status:
                    case LoadStatus.LOAD_SUCCESSFUL:
                        await self._send(State.COMPLETED)
                    case LoadStatus.LOAD_FAILURE | LoadStatus.TIME_MAP_FAILURE | _:
                        await self._send(State.FAILED)
```

**Comment by xjules** [comment]
A question on a general note. Since we are using async context we do not need to worry that forward_model_ok is being executed in job.py. Since the main idea (at least what I [thought is](https://github.com/equinor/ert/issues/6789)) is to execute forward_model_ok for each realization in a sequence from scheduler. But I said async context will take care of it?

**Comment by berland** [comment]
I think we still need to worry, we are executing the done callback in the async context and it will be blocking while it is happening. There are some tricks from the branch `python_jobqueue` to apply at least until we solve it using multiprocessing or threads.

**Comment by xjules** [comment]
You need to rebase but otherwise it is good to go :+1:

**Review by xjules — APPROVED** [review]
Nice job @berland ! :rocket:

**Pull request merged**